src/stream.rs

```diff
@@ -15,19 +16,19 @@ pub enum MaybeHttpsStream<T> {
     /// A stream over plain text.
     Http(T),
     /// A stream protected with TLS.
-    Https(TlsStream<T>),
+    Https(TokioIo<TlsStream<TokioIo<T>>>),
```

Comment:
This `TokioIo<TlsStream<TokioIo<T>>>` business seems pretty ugly to me. Is there some reasonable way we can do this with fewer layers of stuff? I guess we're contrained by what tokio_rustls implements for `TlsStream`?

Reply:
So, our `T` is required to implement the hyper I/O traits so we can delegate them when we implement the I/O traits  for `MaybeHttpsStream`. tokio-rustls' `TlsStream` requires the tokio I/O traits, so we use a `TokioIo` to bridge *from* hyper I/O traits *to* tokio I/O traits. And because the `TlsStream` only implements the tokio traits, we then use another `TokioIo` to bridge to the hyper traits again.

Reply:
In fact that is how hyper-tls does it as well: https://docs.rs/hyper-tls/latest/src/hyper_tls/stream.rs.html#20
Not that pretty but that's the only way to make this work to my knowledge

Reply:
Alright, I guess it's fine for now.